streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/StreamsGraphNode.java

```diff
@@ -70,6 +70,21 @@ public void addParentNode(final StreamsGraphNode parentNode) {
         return new LinkedHashSet<>(childNodes);
     }
 
+    public void clearChildren() {
+        for (final StreamsGraphNode childNode : childNodes) {
+            childNode.removeParent(this);
+        }
+        childNodes.clear();
+    }
+
+    public boolean removeChild(final StreamsGraphNode child) {
+        return childNodes.remove(child) && child.removeParent(this);
+    }
+
+    public boolean removeParent(final StreamsGraphNode parentNode) {
+        return parentNodes.remove(parentNode);
```

Comment:
Since `removeParent` is always called inside removeChild or clearChild only.. actually, that means we can just make `removeParent` a private function right?